### PR TITLE
refactor: replace char_u with char 9: remove `STRCMP`

### PR DESCRIPTION
**Pull request by dundargoc**
Work on https://github.com/neovim/neovim/issues/459

### CODE REVIEW

**Comment by dundargoc** [comment]
Continuation of https://github.com/neovim/neovim/pull/20031

File: src/nvim/eval.c [hunk 1]
```diff
@@ -3996,21 +3996,19 @@ static int get_list_tv(char **arg, typval_T *rettv, int evaluate)
 bool func_equal(typval_T *tv1, typval_T *tv2, bool ic)
 {
   // empty and NULL function name considered the same
-  char_u *s1 =
-    (char_u *)(tv1->v_type == VAR_FUNC ? tv1->vval.v_string : partial_name(tv1->vval.v_partial));
+  char *s1 = (tv1->v_type == VAR_FUNC ? tv1->vval.v_string : partial_name(tv1->vval.v_partial));
```
Comment (zeertzjq) [inline src/nvim/eval.c:3999]:
```suggestion
  char *s1 = tv1->v_type == VAR_FUNC ? tv1->vval.v_string : partial_name(tv1->vval.v_partial);
```

File: src/nvim/eval.c [hunk 2]
```diff
@@ -3996,21 +3996,19 @@ static int get_list_tv(char **arg, typval_T *rettv, int evaluate)
...
   if (s1 != NULL && *s1 == NUL) {
     s1 = NULL;
   }
-  char_u *s2 =
-    (char_u *)(tv2->v_type == VAR_FUNC ? tv2->vval.v_string : partial_name(tv2->vval.v_partial));
+  char *s2 = (tv2->v_type == VAR_FUNC ? tv2->vval.v_string : partial_name(tv2->vval.v_partial));
```
Comment (zeertzjq) [inline src/nvim/eval.c:4003]:
```suggestion
  char *s2 = tv2->v_type == VAR_FUNC ? tv2->vval.v_string : partial_name(tv2->vval.v_partial);
```

File: src/nvim/cmdexpand.c [hunk 3]
```diff
@@ -66,16 +66,16 @@ static int compl_selected;
 
 static int sort_func_compare(const void *s1, const void *s2)
 {
-  char_u *p1 = *(char_u **)s1;
-  char_u *p2 = *(char_u **)s2;
+  char *p1 = *(char **)s1;
+  char *p2 = *(char **)s2;
```
Comment (zeertzjq) [inline src/nvim/cmdexpand.c:70]: The `char_u` on line 2297 should probably also be changed to `char`.

File: src/nvim/grid_defs.h [hunk 4]
```diff
@@ -10,7 +10,7 @@
 #define MAX_MCO  6  // fixed value for 'maxcombine'
 
 // The characters and attributes drawn on grids.
-typedef char_u schar_T[(MAX_MCO + 1) * 4 + 1];
+typedef char schar_T[(MAX_MCO + 1) * 4 + 1];
```
Comment (zeertzjq) [inline src/nvim/grid_defs.h:13]: Changing `schar_` functions to also use `char *` can probably reduce a lot of casts.

File: src/nvim/drawscreen.c [hunk 5]
```diff
@@ -699,42 +699,42 @@ static void win_redr_border(win_T *wp)
   if (adj[0]) {
     grid_puts_line_start(grid, 0);
     if (adj[3]) {
-      grid_put_schar(grid, 0, 0, chars[0], attrs[0]);
+      grid_put_schar(grid, 0, 0, (char_u *)chars[0], attrs[0]);
```
Comment (bfredl) [inline src/nvim/drawscreen.c:702]: `grid_put_schar` should be changed to take a `char *` (along with the other schar functions you changed)

**Comment by bfredl** [comment]
looks pretty much alright, anything missing?

**Comment by dundargoc** [comment]
Nope, good to go.

**Pull request merged**